assembl/static2/js/app/components/cookies/cookiesSelectorContainer.jsx

```diff
@@ -0,0 +1,162 @@
+
+    const cookiesArray = cookiesList && cookiesList.map(
+      cookie => ({ ...this.getCookieObject(cookie), accepted: this.isCookieAccepted(cookie), cookieType: cookie })
+    );
+    const cookiesByCategory = cookiesArray && this.getCookiesObjectFromArray(cookiesArray);
+
+    this.state = {
+      activeKey: 'essential',
+      show: true,
+      cookies: cookiesByCategory
+    };
+  }
+
+  isCookieAccepted = (cookie: string) => {
+    if (cookie.startsWith('ACCEPT')) {
+      return true;
+    }
+    return false;
+  }
+
+  getCookieObject = (cookie: string) => {
+    if (cookie.includes('SESSION_ON_DISCUSSION')) {
+      return { category: 'other', name: COOKIE_TRANSLATION_KEYS.userSession, hasChanged: false };
+    }
+    if (cookie.includes('LOCALE_ON_DISCUSSION')) {
+      return { category: 'essential', name: COOKIE_TRANSLATION_KEYS.locale, hasChanged: false };
+    }
+    if (cookie.includes('TRACKING_ON_DISCUSSION')) {
+      return { category: 'analytics', name: COOKIE_TRANSLATION_KEYS.piwik, hasChanged: false };
+    }
+    return { category: 'other', name: cookie, hasChanged: false };
+  };
+
+  getCookiesObjectFromArray = (cookiesArray: Array<CookieObject>) => {
+    const cookiesObject = {};
+    cookiesArray.forEach(function(cookie) { // eslint-disable-line
+      const { category } = cookie;
+      if (cookiesObject[category]) {
+        cookiesObject[category] = [...cookiesObject[category], cookie];
+      } else {
+        cookiesObject[category] = [cookie];
+      }
+    });
+    return cookiesObject;
+  }
+
+  handleToggle = (updatedCookie: CookieObject) => {
+    const { cookies } = this.state;
+    // Flow bugs with Object.values
+    // $FlowFixMe
+    const cookiesArray:Array<CookieObject> = cookies && Object.values(cookies)
+      .reduce((flat, next) => flat.concat(next), []);
+    const updatedCookiesArray = cookiesArray.map(
+      (cookie: CookieObject) => (cookie.name === updatedCookie.name ? { ...updatedCookie, hasChanged: true } : cookie));
+    const updatedCookiesByCategory = this.getCookiesObjectFromArray(updatedCookiesArray);
+    this.setState({ cookies: updatedCookiesByCategory });
+  }
+
+  toggleCookieType = (cookie: string) =>
+    (cookie.startsWith('ACCEPT') ? cookie.replace('ACCEPT', 'REJECT') : cookie.replace('REJECT', 'ACCEPT'));
+
+
+  saveChanges = () => {
+    const { cookies } = this.state;
+    // Flow bugs with Object.values
+    // $FlowFixMe
+    const cookiesArray:Array<CookieObject> = cookies && Object.values(cookies)
+      .reduce((flat, next) => flat.concat(next), []);
+    const changedCookies = cookiesArray.filter((cookie: CookieObject) => cookie.hasChanged);
+    const changedCookiesNewNames = changedCookies.map(cookie => this.toggleCookieType(cookie.cookieType));
+    const newCookiesList = cookiesArray.filter((cookie: CookieObject) => !cookie.hasChanged).map(cookie => cookie.cookieType);
+    changedCookiesNewNames.forEach((cookie) => { newCookiesList.push(cookie); });
```

Comment:
can't you just do something like : 

```
const newCookies = cookiesArray.map(
    c => c.hasChanged ? this.toggleCookieType(c.cookieType) : c.cookieType
);
```

instead of these 4 lines?

Reply:
You're right, shame on me